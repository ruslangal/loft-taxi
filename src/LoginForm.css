.login-form {
    z-index: 1;
    width: 356px;

    padding: 55px 112px;

    background-color: #fff;
    border-radius: 20px;

    box-shadow: 0px 0px 40px rgba(0, 0, 0, 0.1);
}

.login-form__field {
    display: flex;
    flex-direction: column;

    margin-bottom: 25px;
}

.login-form__forgotten {
    text-align: right;

    font-size: 16px;
    margin-top: 10px;
    color: #828282;
    text-decoration: none;
}

.login-form__title {
    text-align: center;

    margin-bottom: 57px;
}

.login-form__submit {
    background-color: #FDBF5A;

    height: 61px;
    width: 100%;
    margin-top: 46px;

    font-size: 25px;

    border: 0;
    border-radius: 70px;

    cursor: pointer;
}

.login-form__field-input {
    height: 40px;

    font-size: 18px;

    border: 0;
    border-bottom: 2px solid #E4E4E4;
}

.login-form__field-name {
    font-weight: bold;
    font-size: 16px;
}

.login-form__helper {
    margin-top: 33px;

    font-size: 16px;
}

.login-form__helper-link {
    color: #FDBF5A;
    padding-left: 10px;

    text-decoration: none;
}
