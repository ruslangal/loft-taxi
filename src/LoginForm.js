import React, { Component } from 'react'
import './LoginForm.css';

class LoginForm extends Component {
    handleSubmit = (e) => {
        /*
        const userEmail = e.target.userEmail.value;
        const userPassword = e.target.userPassword.value;
        */
        this.props.navigateTo('map');
    }

    handleRegistrationClick = (e) => {
        e.preventDefault();

        this.props.navigateTo('registration');
    }

    render() {
        return (
            <div className="login-form">
                <form className="login-form__form" onSubmit={this.handleSubmit}>
                    <h2 className="login-form__title">Войти</h2>
                    <div className="login-form__field">
                        <span className="login-form__field-name">
                            Email
                        </span>
                        <input name="userEmail" type="text" className="login-form__field-input" placeholder="mail@mail.ru"/>
                    </div>
                    <div className="login-form__field">
                        <span className="login-form__field-name">
                            Пароль
                        </span>
                        <input name="userPassword" type="text" className="login-form__field-input" placeholder="************"/>
                        <a className="login-form__forgotten" href="#" onClick={this.handleRegistrationClick}>
                            Забыли пароль?
                        </a>
                    </div>
                    <input type="submit" value="Войти" className="login-form__submit"/>
                </form>
                <div className="login-form__helper">
                    <span className="login-form__helper-text">Новый пользователь?</span>
                    <a className="login-form__helper-link" href="#" onClick={this.handleRegistrationClick}>Регистрация</a>
                </div>
            </div>
        )
    }
}

export default LoginForm;