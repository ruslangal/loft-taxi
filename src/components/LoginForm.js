import React, { Component } from 'react'
import './LoginForm.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import { auth } from '../store/actions'
import { Link } from 'react-router-dom'

export class LoginFormPresenter extends Component {
    static propTypes = {
        auth: PropTypes.func.isRequired
    }

    handleSubmit = (e) => {
        e.preventDefault();

        const userEmail = e.target.userEmail ? e.target.userEmail.value : null;
        const userPassword = e.target.userPassword ? e.target.userPassword.value : null;

        this.props.auth(userEmail, userPassword);
    }

    render() {
        return (
            <div data-testid="login-form" className="login-form">
                <form className="login-form__form" onSubmit={this.handleSubmit}>
                    <h2 className="login-form__title">Войти</h2>
                    <div className="login-form__field">
                        <span className="login-form__field-name">
                            Email
                        </span>
                        <input name="userEmail" type="text" className="login-form__field-input" placeholder="mail@mail.ru"/>
                    </div>
                    <div className="login-form__field">
                        <span className="login-form__field-name">
                            Пароль
                        </span>
                        <input name="userPassword" type="text" className="login-form__field-input" placeholder="************"/>
                        <Link className="login-form__forgotten" to="/registration">
                            Забыли пароль?
                        </Link>
                    </div>
                    <input data-testid="login-form__submit" type="submit" value="Войти" className="login-form__submit"/>
                </form>
                <div className="login-form__helper">
                    <span className="login-form__helper-text">Новый пользователь?</span>
                    <Link data-testid="login-form__helper-link" className="login-form__helper-link" to="/registration">Регистрация</Link>
                </div>
            </div>
        )
    }
}

export const LoginForm = connect(
    null,
    { auth }
)(LoginFormPresenter);