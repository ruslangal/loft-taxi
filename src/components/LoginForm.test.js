import React from 'react'
import { LoginFormPresenter } from './LoginForm';
import { render, fireEvent, screen } from '@testing-library/react';

describe('LoginFormPresenter', function() {
    it('shows component with correct structure', function() {
        const MockLoginFormPresenter = () => {
            return (
                <LoginFormPresenter
                    navigateTo={(page) => {}}
                    login={(p1, p2) => {}}
                />
            );
        };

        render(<MockLoginFormPresenter />);
        
        expect(screen.getByTestId('login-form')).toMatchSnapshot();
    });

    it('calls login and navigatoTo functions on submit', function() {
        const navigateTo = jest.fn();
        const login = jest.fn();
        const MockLoginFormPresenter = () => <LoginFormPresenter
            navigateTo={navigateTo}
            login={login}
        />;

        render(<MockLoginFormPresenter />);

        fireEvent.click(screen.getByTestId('login-form__submit'));

        expect(login).toHaveBeenCalledTimes(1);
        expect(navigateTo).toHaveBeenCalledTimes(1);
    });

    it('calls navigatoTo function on registration link', function() {
        const navigateTo = jest.fn();
        const MockLoginFormPresenter = () => <LoginFormPresenter
            navigateTo={navigateTo}
            login={() => {}}
        />;
        
        render(<MockLoginFormPresenter />);

        fireEvent.click(screen.getByTestId('login-form__helper-link'));

        expect(navigateTo).toHaveBeenCalledTimes(1);
    });
});
