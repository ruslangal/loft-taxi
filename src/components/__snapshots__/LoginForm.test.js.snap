// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormPresenter shows component with correct structure 1`] = `
<div
  class="login-form"
  data-testid="login-form"
>
  <form
    class="login-form__form"
  >
    <h2
      class="login-form__title"
    >
      Войти
    </h2>
    <div
      class="login-form__field"
    >
      <span
        class="login-form__field-name"
      >
        Email
      </span>
      <input
        class="login-form__field-input"
        name="userEmail"
        placeholder="mail@mail.ru"
        type="text"
      />
    </div>
    <div
      class="login-form__field"
    >
      <span
        class="login-form__field-name"
      >
        Пароль
      </span>
      <input
        class="login-form__field-input"
        name="userPassword"
        placeholder="************"
        type="text"
      />
      <a
        class="login-form__forgotten"
        href="#"
      >
        Забыли пароль?
      </a>
    </div>
    <input
      class="login-form__submit"
      data-testid="login-form__submit"
      type="submit"
      value="Войти"
    />
  </form>
  <div
    class="login-form__helper"
  >
    <span
      class="login-form__helper-text"
    >
      Новый пользователь?
    </span>
    <a
      class="login-form__helper-link"
      data-testid="login-form__helper-link"
      href="#"
    >
      Регистрация
    </a>
  </div>
</div>
`;
